AttitudeManager/Src/AM.cpp

```diff
@@ -2,19 +2,45 @@
 // Created by Anthony Luo on 2022-10-12.
 //
 #include "AM.hpp"
+#include "config.hpp"
 
 #include <array>
 #include <cstdlib>
 
+
+
 namespace AM {
 
+AttitudeManager::AttitudeManager(Flightmode* control_algorithm):
+    control_algorithm_(control_algorithm)
+{
+    for (uint8_t i{0}; i < config::NUM_MOTORS; i++) {
+        
+        motorInstances_[config::motors[i].axis].push_back(
+            {
+                .axis = config::motors[i].axis,
+                .isInverted = config::motors[i].isInverted,
+                .motorInstance = config::motors[i].driverConstructor()
+            }
+        );
```

Comment:
I don't think we necessarily need to copy the values from the config to here. What if we instead store the motorInstances in a normal array, and just create a lookup table of the axis that a motor should be on, and the indices of those motors. That way we don't unnecessarily copy data from the config (since we can just read it from the config) while still retaining the ability to set the motor value in a per-axis fashion?

We could determine the number of motors per axis by iterating over the motor list and counting how many are in each axis. I think this would be fine because we would only need to do it once (during initialization) and the maximum number of motors we can have on zeropilot is 16 (e.g. relatively small).